assembly/integer/i128.ts

```diff
@@ -243,16 +245,17 @@ export class i128 {
 
   @inline @operator('>>')
   static shr(value: i128, shift: i32): i128 {
-    shift &= 127;
+    if (shift == 0) return value;
 
-    if (shift < 64) {
-      var lo = value.lo >>> shift;
-      var hi = value.hi >> shift | (value.lo << (64 - shift));
-      return new i128(lo, hi);
-    } else {
-      var lo = value.hi >> (shift - 64);
-      return new i128(lo, 0);
+    let sign = value.isNeg();
+
+    let res = u128.shr(u128.fromI128(i128.abs(value)),shift).toI128();
+    
+    if (sign) {  // Put back the sign
+      res = res.neg();
```

Comment:
This approach some overhead. For `i128.mul` / `i128.div` this is acceptable, but for such low-level operation like `shr` it is better to operate with the native parts `lo` and `hi` without GC which implicitly produced by `i128.abs`, `toI128` and `res.neg`

Reply:
I didn't manage to make it work another way, I might try again later this week but I don't know if I have the time

Reply:
I didn't have the time to get back on this, I might as well delete shr (and tests) to merge the rest no?

Reply:
That's a good decision